scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/FileOperations.java

```diff
@@ -83,6 +138,13 @@ public void populateDisplayData(Builder builder) {
 
   /** Per-element file reader. */
   public abstract static class Reader<V> implements Serializable {
+    private transient Path path = null;
+
+    public void prepareRead(Path path) throws IOException {
```

Comment:
this method naming makes it a little confusing when to call this `prepareRead` vs the other `prepareRead(ReadableByteChannel channel)`... maybe instead of this method we could rename it like:

```java
private transient Path deletablePath = null;

public void markForCleanup(Path deletablePath) throws IOException {
  this.deletablePath = deletablePath;
}
```